docs/template_plugin/src/template_infer_request.cpp

```diff
@@ -335,7 +335,18 @@ void TemplateInferRequest::inferPostprocess() {
         }
         auto outputBlob = _outputs.at(output.first);
         auto networkOutput = _networkOutputBlobs[output.first];
-        if (outputBlob->getTensorDesc().getPrecision() != networkOutput->getTensorDesc().getPrecision()) {
+        if (result->get_output_partial_shape(0).is_dynamic() &&
+            networkOutput->getTensorDesc().getPrecision() != outputBlob->getTensorDesc().getPrecision() &&
+            networkOutput->getTensorDesc().getPrecision() == Precision::I64 &&
+            outputBlob->getTensorDesc().getPrecision() == Precision::I32) {
+            auto tensor = _outputTensors[_executableNetwork->_outputIndex.at(output.first)];
+            auto* data = new std::int64_t[tensor->get_size_in_bytes() / 8];
+            tensor->read(data, tensor->get_size_in_bytes());
+            auto* write_data = InferenceEngine::as<InferenceEngine::MemoryBlob>(outputBlob)->wmap().as<std::int32_t*>();
+            for (size_t j = 0; j < tensor->get_size_in_bytes() / 8; ++j) {
+                write_data[j] = data[j];
+            }
```

Comment:
Looks like memory leak, because you need deallocate memory after `data` array.

Reply:
Fixed by using a vector instead of a raw pointer.